app/src/main/java/com/babylon/wallet/android/presentation/account/AccountViewModel.kt

```diff
@@ -131,12 +132,16 @@ class AccountViewModel @Inject constructor(
                             }
                         }
                         .onFailure {
-                            _state.update { state ->
-                                state.copy(hasFailedToFetchPricesForAccount = true)
+                            if (it is FiatPriceRepository.PricesNotSupportedInNetwork) {
```

Comment:
It seem a bit odd to me to handle fiat prices absence like this. Shouldn't we have some method/use case to determine if fiat prices are available or not in current context and don't fetch fiat prices at all if we shouldn't? Maybe this way of doing it is outcome of other design choices that I can't understand. What do you think?